drivers/hmis/spec/requests/hmis/project_spec.rb

```diff
@@ -240,6 +231,52 @@
       expect(result.dig('data', 'projects', 'nodes').count).to eq(0)
     end
   end
+
+  describe 'canProjectAcceptReferral query' do
+    let(:can_accept_referral) do
+      <<~GRAPHQL
+        query GetCanProjectAcceptReferral($projectId: ID!, $referrerEnrollmentId: ID!) {
+          canProjectAcceptReferral(projectId:$projectId, referrerEnrollmentId:$referrerEnrollmentId)
+        }
+      GRAPHQL
+    end
+
+    let!(:p2) { create :hmis_hud_project, data_source: ds1, organization: o1, user: u1 }
+    let!(:referral_instance) { create :hmis_form_instance, role: :REFERRAL, entity: p2 }
+
+    let!(:c2) { create :hmis_hud_client, data_source: ds1 }
+
+    # p1 is the referring project
+    let!(:p1_ac) { create_access_control(hmis_user, p1) }
+    let!(:referrer_enrollment) { create :hmis_hud_enrollment, client: c2, project: p1, data_source: ds1 }
+
+    it 'raises an error when the project doesnt accept referrals' do
+      response, result = post_graphql(project_id: p1.id, referrer_enrollment_id: referrer_enrollment.id) { can_accept_referral }
+      expect(response.status).to eq 500
+      expect(result.dig('errors')[0]['message']).to eq 'access denied'
+    end
+
+    it 'raises an error when the current user doesnt have can_manage_outgoing_referrals permission' do
+      remove_permissions(access_control, :can_manage_outgoing_referrals)
+      remove_permissions(p1_ac, :can_manage_outgoing_referrals)
```

Comment:
It's a little hard to tell what's going on with these tests when there are 2 access controls active. Unless we're explicitly trying to test how the access controls behave together, I would change this to only worry about 1 access control. (Or two, if we need to track 1 ACL tied to the source project, and 1 ACL tied to the destination project). In this case the `access_control` is tied to the whole data source, which makes it a little hard to follow.

Since this query no longer has to do with the project query, maybe we can just move this `describe` block to its own spec file? That might make it easier to manage the access controls.